Mathlib/Data/List/Defs.lean

```diff
@@ -749,4 +749,87 @@ Example: if `f : ℕ → list ℕ → β`, `List.mapWithComplement f [1, 2, 3]`
 def mapWithComplement {α β} (f : α → List α → β) : List α → List β :=
   mapWithPrefixSuffix fun pref a suff => f a (pref ++ suff)
 
+/--
+`List.choose` filters list `l` of type `List α` with `chooser` which is of type `α → Option β`,
+yet `β` is ignored.
+It keeps the elements where `chooser x` is `some _` and discards the ones that are `none`
+
+Example:
+```
+def oneTwo? : Nat → Option Nat
+  | 1 => some 1
+  | 2 => some 2
+  | _ => none
+
+[1, 2, 3, 5, 1, 4, 2].choose oneTwo? = [1, 2, 1, 2]
+```
+-/
+def choose (chooser : α → Option β) (l : List α) : List α :=
```

Comment:
As discussed on Zulip, this function is probably better written explicitly and not added as a definition.  Based on the signature, it's easy to mix up this function with `filterMap`.

Reply:
By written explicitly, do you mean use recursion?

Reply:
he means you would use `l.filter (Option.isSome ∘ f)` at the use site instead of `l.choose f`

Reply:
So @gebner is suggesting I remove it. Fair enough. I do find use cases for it sometimes, but it could be easily recreated.

Reply:
The question is, will the definition make it easier for readers to understand code using the definition? If it has a vague name like `choose` they will have to look up the definition, and if the best way to understand what it does is just by combining filter and isSome then the code would have been better off just saying that in the first place and dispensing with the definition.

Reply:
Right. So you just want me to rework the definition and docstring?

Reply:
I think we'd both prefer to keep the `choose` function out of this PR.

Reply:
Ok. I will be sure to take it out on my next commit

Reply:
Deleted.